### PR TITLE
Add endpoint POST/api/articles/:article_id/comments and tests

### PR DESCRIPTION
**Pull request by David-McB**
Add completed endpoint and tests for ticket 7. Also refactored the error handling a bit to hopefully make it cleaner

### CODE REVIEW

File: __tests__/controllers.test.js [hunk 1]
```diff
@@ -105,6 +105,56 @@ describe('GET /api/articles', () => {
...
+                comment_id: 19,
+                body: 'this is a test comment',
+                article_id: 7,
+                author: 'butter_bridge',
+                votes: 0,
+                created_at: expect.any(String)
+            });    
+        });
+    });
+
```
Comment (Dutton1990) [inline __tests__/controllers.test.js:128]: Good happy path test this!

File: __tests__/controllers.test.js [hunk 2]
```diff
@@ -105,6 +105,56 @@ describe('GET /api/articles', () => {
...
+        return request(app).post('/api/articles/23489/comments').send(comment).expect(404)
+        .then(({body}) => expect(body.message).toBe('Error: Key (article_id)=(23489) is not present in table "articles".'));
+    });
+
+    test('should respond with a 400 error if article ID is invalid', () => {
+        const comment = {body: 'test', username: 'butter_bridge'};
+        return request(app).post('/api/articles/notANumber/comments').send(comment).expect(400)
+        .then(({body}) => expect(body.message).toBe('Error: invalid request'));
+    })
+
```
Comment (Dutton1990) [inline __tests__/controllers.test.js:156]: Really nice tests, liking the specific error messages!

**Review by Dutton1990 — CHANGES_REQUESTED** [review]
Great job so far! Two more tests I think you can add. 

201 - ignores unnecessary properties. So if anything else is provided in the request other than username/body it should just be ignored. 
404 - username not found 

Once you've given these a go please resubmit this same PR and we will check!

**Review by NRMSMITH — APPROVED** [review]
Merge away!!! ☄️

**Pull request merged**